cloudsync/api_test.py

```diff
@@ -130,3 +150,148 @@ def test_process_transcode_results(mocker):
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with transaction.atomic():
+        with pytest.raises(ClientError) as exc:
+            process_watch_file(upload)
+            assert 'NoSuchBucket' in str(exc)
+    with pytest.raises(Video.DoesNotExist):
+        Video.objects.get(title=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_filename_error():
+    """Test that a bad filename raises a VideoFilenameError"""
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'Bad filename.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with transaction.atomic():
+        with pytest.raises(VideoFilenameError) as exc:
+            process_watch_file(upload)
+            assert 'No matches found for filename' in str(exc)
+    with pytest.raises(Video.DoesNotExist):
+        Video.objects.get(title=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_process_watch(mocker):
+    """Test that a file with valid filename is processed"""
+    mocker.patch.multiple('cloudsync.tasks.settings',
+                          ET_PRESET_IDS=('1351620000001-000061', '1351620000001-000040', '1351620000001-000020'),
+                          AWS_REGION='us-east-1', ET_PIPELINE_ID='foo')
+    mock_encoder = mocker.patch('cloudsync.api.VideoTranscoder.encode')
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    process_watch_file(upload)
+    mock_encoder.assert_called_once()
+    assert Video.objects.get(title=upload)
+
+
+def test_transcode_job(mocker, video, videofile):
+    """
+    Test that video status is updated properly after a transcode job is successfully created
+    """
+    mocker.patch.multiple('cloudsync.tasks.settings',
+                          ET_PRESET_IDS=('1351620000001-000061', '1351620000001-000040', '1351620000001-000020'),
+                          AWS_REGION='us-east-1', ET_PIPELINE_ID='foo')
+    mock_encoder = mocker.patch('cloudsync.api.VideoTranscoder.encode')
+    transcode_video(video, videofile)  # pylint: disable=no-value-for-parameter
+    mock_encoder.assert_called_once()
+    assert len(video.encode_jobs.all()) == 1
+    assert Video.objects.get(id=video.id).status == VideoStatus.TRANSCODING
```

Comment:
please filter on the status and id here so if the test fails purest would give better log